src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -465,9 +466,16 @@ static int open_device(const char* path, bool try_load_module = false)
     if (r < 0) {
       cerr << "rbd-nbd: failed to load nbd kernel module: " << cpp_strerror(-r) << std::endl;
       return r;
+    } else {
+      load_module = true;
     }
     nbd = open(path, O_RDWR);
   }
+
+  if (nbds_max && try_load_module && !load_module) {
+    cerr << "rbd-nbd: nbds_max param doesn't work, because nbd module already loaded." << std::endl;
```

Comment:
@dillaman Sure, I will create a bug tracker ticket.
@trociny , max_part makes me dilemma, it is different from nbds_max:
the default value in our rbd-nbd is 0, and if(nbds_max) is false if the user not specify nbds_max, however, the default value of max_part is 255(in order to support max partition), and if(max_part) is true, even if the user not specify it...